### ISSUE TITLE
Add warnings when failing to identify a paginated field as a connection

### ISSUE THREAD

**Issue by SorenHolstHansen** [HoudiniGraphql/houdini#222]
I am trying to delete an item from a paginated list, however when i try and add a name to my paginated query, i get the error
`@list can only be applied to types with an id field.`

My schema is like this
```graphql
...

type GroceriesEdge {
  cursor: Int!
  node: Grocery!
}

type Grocery {
  id: Int!
  name: String!
}

type PaginatedGroceries {
  edges: [GroceriesEdge!]!
  pageInfo: PageInfo!
}
```

And i am writing my query like this
```js
const { loadNextPage, data } = paginatedQuery<Groceries>(graphql`
		query Groceries {
			paginatedGroceries(first: 10) @paginate(name: "SomeName") {
				edges {
					node {
						id
						name
					}
				}
			}
		}
	`);
```
But then when I generate the types i get the mentioned error.

I looked at the example in the repo, and I don't see where my schema is different from yours.

Do you know what I am doing wrong?

**Comment by AlecAivazis** [comment]
Hm, this definitely looks right - do you mind sharing a link to a repo I can try to reproduce this on? Thanks for being patient btw, all of my time has been commited to a few exciting changes for houdini

**Comment by SorenHolstHansen** [comment]
I made a failing example [here](https://github.com/SorenHolstHansen/houdini-lists-error/blob/main/README.md). 
No worries, looking very much forward to the new Houdini changes, and this is by no means urgent

**Comment by AlecAivazis** [comment]
Hey @SorenHolstHansen - I finally found some time to dig into your issue. Looks like the problem stems from your `paginatedGroceries` field. It doesn't conform to the cursor-based connection spec: `first` needs  to be an `Int`, not a `Float`. This makes houdini struggle to identify it as a valid connection list so it's looking for an `id` field on the connection type, not the actual element of the list.

**Comment by AlecAivazis** [comment]
The generator should have absolutely detected that the field existed but with the wrong type and given you a more useful warning message. [This](https://github.com/HoudiniGraphql/houdini/blob/39065e5ad649280c560c9da50f393340b67413a5/packages/houdini/cmd/transforms/list.ts#L277) is the function that's responsible for identifying a field as a connection. Assuming everything works, I'll update the issues title to reflect that we need to improve the messaging here

**Comment by AlecAivazis** [comment]
This is now fixed in #344
